tests/test_transforms.py

```diff
@@ -96,3 +96,73 @@ def test_tth_to_q(inputs, expected):
 def test_tth_to_q_bad(inputs, expected):
     with pytest.raises(expected[0], match=expected[1]):
         tth_to_q(inputs[1], inputs[0])
+
+
+params_q_to_d = [
+    # UC1: User specified empty q values
+    ([np.array([])], np.array([])),
+    # UC2: User specified valid q values
+    (
+        [np.array([np.pi / 6, 1 * np.pi, 2 * np.pi, 3 * np.pi, 4 * np.pi, 5 * np.pi])],
+        np.array([0.4, 0.5, 0.66667, 1, 2, 12]),
```

Comment:
shall we print a warning message if d is too small / large? 12 might not be very extreme here, but if we have q~0 we might get a really large d. i think it doesn't really make sense if we change that d value, but just warn the users that we get a value that may cause trouble.

Reply:
no I don't think so.

Has the d-array become inverted?  Shouldn't it go from large to small?